bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -275,14 +340,18 @@ private void refreshAll() {
         // Determine whether the current position is one of these illegal guiders.  If so, we add
         // the guide probe to the list of choices so that this target may be selected in order to
         // change its type or delete it.
-        GuideProbe illegal = null;
-        for (final GuideProbe guider : illegalSet) {
-            final Option<GuideProbeTargets> gtOpt = env.getPrimaryGuideProbeTargets(guider);
-            if (gtOpt.getValue().getTargets().contains(_curPos)) {
-                illegal = guider;
-                guiders.add(guider);
-            }
-        }
+        final Option<GuideProbe> illegal = _curSelection.flatMap(either -> either.foldLeft(t -> {
+            final Option<GuideProbe> noProbe = None.instance();
+            return illegalSet.stream().sequential().reduce(noProbe,
+                    (illegalOpt,gp) -> {
+                        final Option<GuideProbeTargets> gtOpt = env.getPrimaryGuideProbeTargets(gp);
+                        if (gtOpt.getValue().getTargets().contains(t)) {
+                            guiders.add(gp);
+                            return new Some<>(gp);
+                        }
+                        else return illegalOpt;
+                    }, (gp1,gp2) -> gp1);
+        }).getOrElse(None.instance()));
```

Comment:
It didn't particularly need to change, but the behaviour should be the same (I hope?) and by using Java 8 functional programming and lambdas, I could eliminate the non-final variables.